cloud/gcp/storage/main.tf

```diff
@@ -0,0 +1,75 @@
+resource "google_kms_key_ring" "keyring" {
+  name     = "civform-keyring"
+  location = var.region
+}
+
+resource "google_kms_crypto_key" "storage-key" {
+  name            = "cloud-storage-key"
+  key_ring        = google_kms_key_ring.keyring.id
+  rotation_period = "604800s" //7 days
+
+  lifecycle {
+    prevent_destroy = true
+  }
+}
+
+// Need the gcs service account name beforehand.
+data "google_storage_project_service_account" "gcs_account" {
+}
+
+// Allow gcs service account on key.
+resource "google_kms_crypto_key_iam_binding" "kms-binding" {
+  crypto_key_id = google_kms_crypto_key.storage-key.id
+  role          = "roles/cloudkms.cryptoKeyEncrypterDecrypter"
+  members = ["serviceAccount:${data.google_storage_project_service_account.gcs_account.email_address}"]
+}
+
+
+resource "google_storage_bucket" "file-storage" {
+  name          = "cloud-filestore-${var.company-name}"
+  location      = var.region
+  force_destroy = true
```

Comment:
question (blocking): what does `force_destroy` do?

Reply:
it will delete the bucket if I run tf destroy

Reply:
ah great